python_modules/libraries/dagstermill/dagstermill/manager.py

```diff
@@ -298,15 +303,28 @@ def yield_result(self, value, output_name="result"):
                 f"Expected one of {[str(output_def.name) for output_def in self.solid_def.output_defs]}"
             )
 
-        dagster_type = self.solid_def.output_def_named(output_name).dagster_type
-
-        # https://github.com/dagster-io/dagster/issues/2648
-        # dagstermill temporary file creation should use a more systematic and robust scheme
-        out_file = os.path.join(
-            self.marshal_dir, f"{self.context.solid_handle}-output-{output_name}"
+        # pass output value cross process boundary using io manager
+        step_context = self.context._step_context  # pylint: disable=protected-access
+        # Note: yield_result currently does not support DynamicOutput
+        step_output_handle = StepOutputHandle(
+            step_key=step_context.step.key, output_name=output_name
         )
+        output_context = step_context.get_output_context(step_output_handle)
+        io_manager = step_context.get_io_manager(step_output_handle)
+
+        io_manager.handle_output(output_context, value)
 
-        scrapbook.glue(output_name, write_value(dagster_type, value, out_file))
+        # warn when we detect the io manager is asymmetric and discourage it
+        value_from_load_input = io_manager.load_input(
```

Comment:
Are we only loading this value for the sake of raising the below warning? I think it's not worth the performance hit - in some situations, loading could be very expensive, and we're incurring enough overhead already.

Reply:
yes - will remove